crates/ide_assists/src/handlers/replace_qualified_name_with_use.rs

```diff
@@ -84,7 +84,8 @@ pub(crate) fn replace_qualified_name_with_use(
                 ImportScope::Module(it) => ImportScope::Module(builder.make_mut(it)),
                 ImportScope::Block(it) => ImportScope::Block(builder.make_mut(it)),
             };
-            shorten_paths(scope.as_syntax_node(), &path.clone_for_update());
+            shorten_paths(scope.as_syntax_node(), &path);
```

Comment:
It seems like only `scope` is mutated. `path` is only used in `path_eq_no_generics` in an immutable fashion.